app-backend/datamodel/src/main/scala/com/azavea/rf/datamodel/ColorCorrect.scala

```diff
@@ -401,45 +392,35 @@ object ColorCorrect {
           }
         }
 
-    val maybeAdjustBrightness =
-      for (b <- params.brightness)
-        yield (mb: MultibandTile) => mb.mapBands { (i, tile) => adjustBrightness(tile, b) }
-
-    val maybeAdjustContrast =
-      for (c <- params.contrast)
-        yield (mb: MultibandTile) => mb.mapBands { (i, tile) => adjustContrast(tile, c) }
-
     val maybeAdjustGamma =
-      for (c <- params.contrast)
+      for (c <- params.sigmoidalContrast.alpha)
         yield (_: MultibandTile).mapBands { (i, tile) =>
           i match {
             case 0 =>
-              params.redGamma.fold(tile)(gamma => gammaCorrect(tile, gamma))
+              params.gamma.redGamma.fold(tile)(gamma => gammaCorrect(tile, gamma))
```

Comment:
I refactored this to use explicit `match` statements and although it is a lot more verbose, it's significantly clearer.